cowbird/database/base.py

```diff
@@ -0,0 +1,59 @@
+import abc
+from typing import TYPE_CHECKING
+
+from cowbird.database.stores import StoreInterface
+
+if TYPE_CHECKING:
+    from typing import Type, Union
+
+    from cowbird.typedefs import JSON, AnySettingsContainer
+    StoreSelector = Union[Type[StoreInterface], StoreInterface, str]
+
+
+class DatabaseInterface(metaclass=abc.ABCMeta):
+    """
+    Return the unique identifier of db type matching settings.
+    """
+    __slots__ = ["type"]
+
+    def __init__(self, _):
+        # type: (AnySettingsContainer) -> None
+        if not self.type:  # pylint: disable=E1101,no-member
+            raise NotImplementedError("Database 'type' must be overridden in inheriting class.")
+
+    @staticmethod
+    def _get_store_type(store_type):
+        # type: (StoreSelector) -> str
+        if isinstance(store_type, StoreInterface):
+            return store_type.type
+        if isinstance(store_type, type) and issubclass(store_type, StoreInterface):
```

Comment:
Is this check necessary? My understanding of `isinstance()` is that it returns True for class and/or subclass, which would make this check redundant (unless there's a specific case I'm not familiar with).

If it's indeed necessary, since it share's the same `return` with the previous check, I'd be more inclined to combine them.

Reply:
store_type can be any of those : 

1. StoreInterface instance; given `store_type =  MonitoringStore(...)`, `isinstance(store_type, StoreInterface)` return `True`
2. Class type subclassing StoreInterface class; given `store_type = MonitoringStore`, `isinstance(store_type, type)` return `True` because `MonitoringStore` is a `type`, not an instance. And `issubclass(store_type, StoreInterface)` return true because the `MonitoringStore` is a type subclassing `StoreInterface`
3. A string; (no issue there)

They return the same thing but in the first case it resolves to self.type, while in the second case it resolves to cls.type. I could add an explanation above those ifs.

Reply:
Ok, perfect. I didn't anticipate it not being used on a class instance.